### PR TITLE
Update dskit to support gzip-compressed runtime configs

### PR DESCRIPTION
**Pull request by colega**
#### What this PR does

This allows larger runtime configuration files without hitting the Kubernetes ConfigMap size limit.

Implemented in dskit in https://github.com/grafana/dskit/pull/571

#### Which issue(s) this PR fixes or relates to

Fixes N/A

#### Checklist

- [x] Tests updated (tested in dskit).
- [x] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Review by pstibrany — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -71,6 +71,7 @@
 * [ENHANCEMENT] Make `-query-frontend.additional-query-queue-dimensions-enabled` and `-query-scheduler.additional-query-queue-dimensions-enabled` non-operational flags in preparation for removal. #8984
 * [ENHANCEMENT] Add a new ingester endpoint to prepare instances to downscale. #8956
 * [ENHANCEMENT] Query-scheduler: Add `query-scheduler.prioritize-query-components` which, when enabled, will primarily prioritize dequeuing fairly across queue components, and secondarily prioritize dequeuing fairly across tenants. When disabled, tenant fairness is primarily prioritized. `query-scheduler.use-multi-algorithm-query-queue` must be enabled in order to use this flag. #9016 #9071
+* [ENHANCEMENT] Runtime configuration can now read gzip-compressed files with `.gz` extension. #9074
```
Comment (tacole02) [inline CHANGELOG.md:74]:
```suggestion
* [ENHANCEMENT] Update runtime configuration to read gzip-compressed files with `.gz` extension. #9074
```

File: docs/sources/mimir/configure/about-runtime-configuration.md [hunk 2]
```diff
@@ -23,6 +23,8 @@ Runtime configuration values take precedence over command-line options.
 
 If multiple runtime configuration files are specified the runtime configuration files will be merged in a left to right order.
 
+If a runtime configuration file has the `.gz` extension, it will be decompressed before being read.
```
Comment (tacole02) [inline docs/sources/mimir/configure/about-runtime-configuration.md:26]:
```suggestion
If a runtime configuration file has the `.gz` extension, it's decompressed before being read.
```

Comment (tacole02) [reply docs/sources/mimir/configure/about-runtime-configuration.md:26]: We try to avoid using "will" in docs and instead always opt for the present tense.

Comment (tacole02) [reply docs/sources/mimir/configure/about-runtime-configuration.md:26]: Is there a way we can use active voice here without losing the meaning? If not, it's okay as-is.

**Review by tacole02 — APPROVED** [review]
Thank you 😄

File: docs/sources/mimir/configure/about-runtime-configuration.md (hunk 2 above)
Comment (colega) [reply docs/sources/mimir/configure/about-runtime-configuration.md:26]: Okay I'll do this active voice:
```suggestion
Add a `.gz` extension to the configuration files that should be decompressed before parsing YAML.
```
Thank you. 

I'm going to merge it with this, but please don't hesitate to provide extra suggestions, I can update the docs in a separate PR later.

**Pull request merged**